File: Week_0/7_Function_Overloading.py
class Calculator:
    def sum(self, num1, num2, num3 = None):
       if num3 is None:
           # to sum two numbers
           return num1+num2  
       # method overloading to sum three numbers   
       return num1+num2+num3    
    
    
cal = Calculator()

print(cal.sum(1,2,3))
print(cal.sum(1,2))

File: Week_0/7_Function_Overloading.cpp
#include <bits/stdc++.h>
using namespace std;

// Calculator class
class Calculator {
public:
    // Method to sum two numbers
    int sum(int num1, int num2) {
        return num1 + num2;
    }

    // Methods overloading to sum three numbers
    int sum(int num1, int num2, int num3) {
        return num1 + num2 + num3;
    }
};


int main()
{
    // Creating a Calculator object
    Calculator cal;

    // Call with three arguments
    cout << cal.sum(1, 2, 3) << endl;

    // Call with two arguments
    cout << cal.sum(1, 2) << endl;
    

    return 0;
}
